src/cloudimagedirectory/transform/transform.py

```diff
@@ -295,27 +294,143 @@ def run(self, data: type[Transformer]) -> list:
+
+                results.append(data_entry)
+        return results
+
+
+class TransformerGoogleV2RHEL(Transformer):
+    """Transform raw rhel Google data into the schema."""
+
+    def run(self, data: list) -> list:
+        """Transform the raw data."""
+        # NOTE: Verify that the data is raw and provided by google.
+        entries = [x for x in data if x.is_provided_by("google") and x.is_raw()]
+
+        results = []
+        for e in entries:
+            entry = copy.deepcopy(e)
+            raw = self.src_conn.get_content(entry)
+
+            for content in raw.content:
+                content["creation_timestamp"] = content["creationTimestamp"]
+                if "rhel" in content["name"]:
+                    image_data = format_google.image_rhel(content)
+                    image_name = image_data["name"].replace(" ", "_").lower()
+                    region = "global"
+                    os_name = "rhel"
+                    provider = "google"
+                    version = image_data["version"]
+                    # NOTE: Due to consistency issues between the cloud providers and the fact
+                    # that they do not all have unique numbers to identify their images, we decided
+                    # to use this solution instead.
+                    image_id = hashlib.sha1(image_name.encode()).hexdigest()  # noqa: S324
+
+                    # NOTE: example of expected paths
+                    # v2/os/rhel/provider/google/version/8.6.0/region/global/image/71d0a7aaa1f0dc06840e46f6ce316a7acfb022d4
+                    # v2/os/rhel/provider/google/version/8.2.0/region/global/image/14e4eab326cc5a2ef13cb5c0f36bc9bfa41025d9
+                    path = f"v2/os/{os_name}/provider/{provider}/version/{version}/region/{region}/image/{image_id}"
+                    data_entry = connection.DataEntry(path, image_data)
+
+                    results.append(data_entry)
+        return results
+
+
 class TransformerV2All(Transformer):
     """Genearate list of all image details."""
 
     # TODO: Mypy says that 'data' below is not iterable. This needs to be fixed later.
     @no_type_check
     def run(self, data: type[Transformer]) -> list:
-        """Sort the raw data."""
-        # NOTE: Verify that the data is not raw.
-        entries = [x for x in data if not x.is_raw() and not x.is_provided_by("idx")]
+        # NOTE: Verify that the data is from api v2.
+        entries = [x for x in data if x.is_API("v2")]
 
         results = []
 
         for e in entries:
             entry = copy.deepcopy(e)
+
             filename = entry.filename.split("/")
             if len(filename) < 3:
                 print("warn: could not determine region or provider of image: " + entry.filename)
                 continue
 
-            entry.content["provider"] = filename[1]
-            entry.content["region"] = filename[2]
+            entry.content["provider"] = filename[4]
```

Comment:
This is totally fine but I can imagine that these array access values might be confusing in the long run. Maybe we can use namedtubles in the future? It would at least make accessing the individual fields easier.

Just an example:

```
  from collections import namedtuple
  
  APIPath = namedtuple('APIPath', ['api_version', 'distro', 'provider', 'version','region','image_id'])
  
  split_path = "v2/rhel/google/9.0.0/us-east-1/asdasdasdadaf".split('/')
  api_path = APIPath(*split_path)
  
  print(api_path.api_version)
  print(api_path.image_id)
```
That api_path could be a property of the DataEntry object or something.

Reply:
I agree, that looks way better to maintain. :smile: Is it ok, when I adapt in another pr? :)

Reply:
I created a new issue for it :) https://github.com/redhatcloudx/transformer/issues/685

Reply:
Absolutely, thanks for taking care of this.